app/src/main/java/com/io/moviO/movieDetails/MovieDetailsViewModel.kt

```diff
@@ -3,17 +3,27 @@ package com.io.moviO.movieDetails
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.io.moviO.data.DataResult
 import com.io.moviO.data.Movie
 import com.io.moviO.domain.GetMovieByIdUseCase
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
 class MovieDetailsViewModel : ViewModel() {
 
-    private val _movie = MutableLiveData<Movie>()
-    val movie: LiveData<Movie> = _movie
+    private val _movie = MutableLiveData<DataResult<Movie>>()
+    val movie: LiveData<DataResult<Movie>> = _movie
 
     private val useCase = GetMovieByIdUseCase()
 
     fun getMovieById(id: String) {
-        _movie.value = useCase.getMovieById(id)
+        viewModelScope.launch(Dispatchers.IO) {
+            val result = useCase.execute(id)
+            withContext(Dispatchers.Main) {
```

Comment:
look up the difference between setValue and postValue, if you use the right one you would't need `withContext(Dispatchers.Main)`

Reply:
setValue() method must be called from the main thread. But if you need set a value from a background thread, postValue() should be used.